File: Python/150. Evaluate Reverse Polish Notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        st = []
        for a in tokens:
            if a == '+' or a == '-' or a == '*' or a == '/':
                num2 = st.pop(-1)
                num1 = st.pop(-1)
                if a == '+': st.append(num1 + num2)
                if a == '-': st.append(num1 - num2)
                if a == '*': st.append(num1 * num2)
                if a == '/': st.append(int(num1 / num2))
            else:
                st.append(int(a))
        return st[-1]

File: C++/150. Evaluate Reverse Polish Notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> st;
        for (auto a : tokens) {
            if (a == "+" || a == "-" || a == "*" || a == "/") {
                int num2 = st.top(); st.pop();
                int num1 = st.top(); st.pop();
                if (a == "+") st.push(num1 + num2);
                if (a == "-") st.push(num1 - num2);
                if (a == "*") st.push(num1 * num2);
                if (a == "/") st.push(num1 / num2);
            }
            else {
                st.push(stoi(a));
            }
        }
        return st.top();
    }
};